src/test/regress/sql/grant_on_schema_propagation.sql

```diff
@@ -0,0 +1,153 @@
+\c - - - :master_port
+
+-- grant with multiple permissions, roles and schemas
+GRANT USAGE, CREATE ON SCHEMA dist_schema, another_dist_schema, non_dist_schema TO role_1, role_2, role_3;
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname IN ('dist_schema', 'another_dist_schema', 'non_dist_schema') ORDER BY nspname;
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname IN ('dist_schema', 'another_dist_schema', 'non_dist_schema') ORDER BY nspname;
+\c - - - :master_port
+
+-- revoke with multiple permissions, roles and schemas
+REVOKE USAGE, CREATE ON SCHEMA dist_schema, another_dist_schema, non_dist_schema FROM role_1, role_2;
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname IN ('dist_schema', 'another_dist_schema', 'non_dist_schema') ORDER BY nspname;
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname IN ('dist_schema', 'another_dist_schema', 'non_dist_schema') ORDER BY nspname;
+\c - - - :master_port
+
+-- grant with grant option
+GRANT USAGE ON SCHEMA dist_schema TO role_1, role_3 WITH GRANT OPTION;
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname IN ('dist_schema', 'another_dist_schema', 'non_dist_schema') ORDER BY nspname;
+\c - - - :master_port
+
+-- revoke grant option for
+REVOKE GRANT OPTION FOR USAGE ON SCHEMA dist_schema FROM role_3;
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname IN ('dist_schema', 'another_dist_schema', 'non_dist_schema') ORDER BY nspname;
+\c - - - :master_port
+
+-- test current_user
+SET citus.enable_alter_role_propagation TO ON;
+ALTER ROLE role_1 SUPERUSER;
+SET citus.enable_alter_role_propagation TO OFF;
+SET ROLE role_1;
+GRANT CREATE ON SCHEMA dist_schema TO CURRENT_USER;
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname IN ('dist_schema', 'another_dist_schema', 'non_dist_schema') ORDER BY nspname;
+\c - - - :master_port
+RESET ROLE;
+SET citus.enable_alter_role_propagation TO ON;
+ALTER ROLE role_1 NOSUPERUSER;
+SET citus.enable_alter_role_propagation TO OFF;
+
+SELECT run_command_on_coordinator_and_workers('DROP SCHEMA dist_schema, another_dist_schema, non_dist_schema CASCADE');
+
+-- test if the grantors are propagated correctly
+-- first remove one of the worker nodes
+SET citus.shard_replication_factor TO 1;
+SELECT master_remove_node('localhost', :worker_2_port);
+
+-- create a new schema
+CREATE SCHEMA grantor_schema;
+
+-- give cascading permissions
+GRANT USAGE, CREATE ON SCHEMA grantor_schema TO role_1 WITH GRANT OPTION;
+GRANT CREATE ON SCHEMA grantor_schema TO PUBLIC;
+SET ROLE role_1;
+GRANT USAGE ON SCHEMA grantor_schema TO role_2 WITH GRANT OPTION;
+GRANT CREATE ON SCHEMA grantor_schema TO role_2;
+GRANT USAGE, CREATE ON SCHEMA grantor_schema TO role_3;
+GRANT CREATE, USAGE ON SCHEMA grantor_schema TO PUBLIC;
+SET ROLE role_2;
+GRANT USAGE ON SCHEMA grantor_schema TO role_3;
```

Comment:
doesn't `role_3` already have usage granted here from 3 lines up? Maybe remove the grant from there?

Reply:
This is to check if grants from different grantors are represented correctly.

Reply:
makes sense